CDP4Reporting/ViewModels/ReportDesignerViewModel.cs

```diff
@@ -0,0 +1,935 @@
+        /// </summary>
+        /// <param name="parameters">The <see cref="IEnumerable{IReportingParameter}"/>, which could be used to create a filter string</param>
+        /// <returns>The create Filter string in <see cref="IReportingParameters.CreateFilterString"/></returns>.
+        private string GetFilterString(IEnumerable<IReportingParameter> parameters)
+        {
+            if (this.CompileResult == null)
+            {
+                this.AddOutput("Compile data source code first.");
+                return null;
+            }
+
+            AppDomain.CurrentDomain.AssemblyResolve += this.AssemblyResolver;
+
+            try
+            {
+                var editorFullClassName =
+                    this.CompileResult
+                        .CompiledAssembly
+                        .GetTypes()
+                        .FirstOrDefault(t => t.GetInterfaces()
+                            .Any(i => i == typeof(IReportingParameters))
+                        )?.FullName;
+
+                if (editorFullClassName == null)
+                {
+                    return null;
+                }
+
+                if (!(this.CompileResult.CompiledAssembly.CreateInstance(editorFullClassName) is IReportingParameters instObj))
+                {
+                    this.AddOutput("Report parameter class not found.");
+                    return null;
+                }
+
+                return instObj.CreateFilterString(parameters);
+            }
+            catch (Exception ex)
+            {
+                this.AddOutput(ex.ToString());
+            }
+            finally
+            {
+                AppDomain.CurrentDomain.AssemblyResolve -= this.AssemblyResolver;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Needed for using the CDP4Reporting assembly
+        /// </summary>
+        /// <param name="sender">The sender <see cref="object"/></param>
+        /// <param name="args">The <see cref="ResolveEventArgs"/></param>
+        /// <returns></returns>
+        private Assembly AssemblyResolver(object sender, ResolveEventArgs args)
+        {
+            if (args.Name == this.GetType().Assembly.FullName)
+            {
+                return this.GetType().Assembly;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Add text to the output pane
+        /// </summary>
+        /// <param name="text">The text</param>
+        private void AddOutput(string text)
+        {
+            this.Output += $"{DateTime.Now:HH:mm:ss} {text}{Environment.NewLine}";
+        }
+
+        /// <summary>
+        /// Get report zip archive components
+        /// </summary>
+        /// <param name="rep4File">archive zip file</param>
+        /// <returns>The <see cref="ReportZipArchive"/></returns>
+        private ReportZipArchive GetReportStream(string rep4File)
```

Comment:
i have a strong preference for explicit names that are easy to understand 1 year or 2 years from now. Please update this to be a word that conveys the meaning

Reply:
I renamed the method to GetReportZipArchive.